articles/200_migration_guide_from_ros1.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: default
+title: Migration guide from ROS 1
+permalink: articles/migration_guide_from_ros1.html
+abstract:
+  This article describes the steps to migrate a ROS 1 package to ROS 2.
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+
+## Prerequisite
+
+Before being able to migrate a ROS 1 package to ROS 2 all of its dependencies must be available in ROS 2.
+
+
+## Migration steps
+
+### Package naming
+
+Packages containing messages and / or services which should be supported by the [ROS bridge](https://github.com/ros2/ros1_bridge/) need to have a different name in ROS 1 and ROS 2.
+Therefore the suffixes `_msgs` and `_srvs` are replaced (and merged in a single package) with the suffix `_interfaces`.
+
+
+### Package manifests
+
+ROS 2 only support the format 2 of the package specification which is defined in [REP 140](http://www.ros.org/reps/rep-0140.html).
+Therefore the `package.xml` file must be update to format 2 if it uses format 1.
+Since ROS 1 support both formats (1 as well as 2) it is save to perform that conversion in the ROS 1 package.
+
+Some packages might have different names in ROS 2 so the dependencies might need to be updated accordingly.
+
+
+### Messages and services
+
+Message files ending in `.msg` must be located in the subfolder `msg`.
+Service files ending in `.srv` must be located in the subfolder `srv`.
+
+These files might need to be updated to comply with the [ROS Interface definition](http://design.ros2.org/articles/interface_definition.html).
+Commonly that involves updating the definition of `Duration` and `Time` fields to the new builtin messages.
```

Comment:
I think this could be clearer, maybe: "... involves updating your files to account for the fact that in ROS 1 `time` and `duration` are built-in primitive types but in ROS 2 they are the normal messages `Time` and `Duration`."

I agree with Jackie, and I would suggest that maybe we should set this up as a list of known changes that are commonly made, rather than a single statement, and extend it as we go. Linking to the above article is good, but that states what the rules are in ROS 2, not what is different from ROS 1.